network/proxy/mucp/mucp.go

```diff
@@ -342,13 +329,41 @@ func NewProxy(opts ...options.Option) proxy.Proxy {
 		p.Client = c.(client.Client)
 	}
 
+	// set the default client
+	if p.Client == nil {
+		p.Client = client.DefaultClient
+	}
+
 	// get router
 	r, ok := p.Options.Values().Get("proxy.router")
 	if ok {
 		p.Router = r.(router.Router)
-		// TODO: should we advertise?
-		go p.Router.Advertise()
 	}
 
+	// create default router and start it
+	if p.Router == nil {
```

Comment:
Looking at the getRoute method now, it seems as though p.Router interface is never nil, so it will always use that and in the event MICRO_ROUTER or MICRO_ROUTER_ADDRESS env vars are set these are basically ignored unless p.Router returns no routes. 

Previous logic was:

1. If p.Router interface exists use it
2. If MICRO_ROUTER/MICRO_ROUTER_ADDRESS do not exist, set the default router interface and use that
3. If MICRO_ROUTER or MICRO_ROUTER_ADDRESS are present then use those

So think thats a bug with the logic. Might be that having both interface and RPC use in the same code path complicates things. Alternative would be to only use the interface and separately implement a Router interface that makes use of RPC. Fixing the original logic will keep us progressing, doing the second as a refactor just seems excessive right now.

What do you think?

Reply:
Ah, I always forget about the preference of the `env` variables in `micro`. This whole mixing and matching of the config, env vars etc creates so much `if` jumps in `go-micro` package code; it's not easy to navigate. Or maybe it's just me.

Either way, this is a great catch wrt `micro` way of dealing with env and config stuff. It makes me think we should have some sort of glue code which handles this separately so that this logic does not leak into every single piece of code we write or something 🤔

Reply:
I need to have a think about this one. `MICRO_ROUTER` and `MICRO_ROUTER_ADDRESS` have the absolute precedence I take it. I need to come up with some non-hacky/crufty way of doing that in code.